test/e2e/framework/pods.go

```diff
@@ -146,6 +149,27 @@ func (c *PodClient) Update(name string, updateFn func(pod *v1.Pod)) {
 	}))
 }
 
+// AddEphemeralContainerSync adds an EphemeralContainer to a pod and waits for it to be running.
+func (c *PodClient) AddEphemeralContainerSync(pod *v1.Pod, ec *v1.EphemeralContainer, timeout time.Duration) {
+	namespace := c.f.Namespace.Name
+
+	podJS, err := json.Marshal(pod)
+	ExpectNoError(err, "error creating JSON for pod: %v", err)
+
+	ecPod := pod.DeepCopy()
+	ecPod.Spec.EphemeralContainers = append(ecPod.Spec.EphemeralContainers, *ec)
+	ecJS, err := json.Marshal(ecPod)
+	ExpectNoError(err, "error creating JSON for pod with ephemeral container: %v", err)
+
+	patch, err := strategicpatch.CreateTwoWayMergePatch(podJS, ecJS, pod)
+	ExpectNoError(err, "error creating patch to add ephemeral container: %v", err)
+
+	_, err = c.Patch(context.TODO(), pod.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{}, "ephemeralcontainers")
+	ExpectNoError(err, "Failed to patch ephemeral containers in pod %q: %v", pod.Name, err)
```

Comment:
Maybe use `format.Pod(pod)` instead of the name

```suggestion
	ExpectNoError(err, "Failed to patch ephemeral containers in pod %q: %v", format.Pod(pod), err)
```

Reply:
nice tip for format.Pod, thanks.